modules/ctl/README.md

```diff
@@ -0,0 +1,197 @@
+Are you sure you want to continue connecting (yes/no)? yes
+Warning: Permanently added '192.168.99.100' (ECDSA) to the list of known hosts.
+docker@192.168.99.100's password:
+$ _
+```
+
+Verify that you can `ssh` into the VM.
+
+```
+$ ssh docker@192.168.99.100
+                        ##         .
+                  ## ## ##        ==
+               ## ## ## ## ##    ===
+           /"""""""""""""""""\___/ ===
+      ~~~ {~~ ~~~~ ~~~ ~~~~ ~~~ ~ /  ===- ~~~
+           \______ o           __/
+             \    \         __/
+              \____\_______/
+ _                 _   ____     _            _
+| |__   ___   ___ | |_|___ \ __| | ___   ___| | _____ _ __
+| '_ \ / _ \ / _ \| __| __) / _` |/ _ \ / __| |/ / _ \ '__|
+| |_) | (_) | (_) | |_ / __/ (_| | (_) | (__|   <  __/ |
+|_.__/ \___/ \___/ \__|_____\__,_|\___/ \___|_|\_\___|_|
+Boot2Docker version 17.03.0-ce, build HEAD : f11a204 - Thu Mar  2 00:14:47 UTC 2017
+Docker version 17.03.0-ce, build 3a232c8
+docker@default:~$ exit
+Connection to 192.168.99.100 closed.
+$ _
+```
+
+You can do `docker-machine stop` to shut down the VM and `docker-machiner start` to start it back up again. Anyway we're now all set for a "remote" deployment.
+
+## Deploying and Upgrading Gem
+
+Ok so `gemctl` is an alias in the build that lets you run deployment stuff from the sbt prompt. If you just type `gemctl` at the sbt prompt you'll get a help message. `gemctl -h` will list available commands, and `gemctl <command> -h` will show details on any specific command. If the help messages aren't helpful, please improve them.
+
+#### Packaging the Application
+
+In order to deply the app we must first create a **docker image** of our application. We do this via sbt-native-packager as follows:
+
+```
+> telnetd/docker:publishLocal
+```
+
+Image names have the form `<repo>/<artifact>:<version>`. Ours will be `geminihlsw/gem-telnetd:<version>` where the version is the git commit hash, or the hash followed by `-UNCOMMITTED` if there are uncommitted changes. This give us very fine-grained versioning and a reliable partial ordering to know that an upgrade is legal. Note that if the filesystem version isn't the same as the version computed by sbt you'll need to reload the project. The prompt will include a message to this effect.
+
+The image will be stored in docker's internal storage area, which shares "slices" with other images built from the same base to save space.
+
+For remote deployments we must publish and push the image to a repository where it can be seen by other machines. For now we're using docker-hub but it probably makes sense to set up an internal repository. To do this you need to create a docker-hub account and be added to the `geminihlsw` organization (talk to Rob), and then you can do `telnetd/docker:publish` to push the image up to the cloud. Note that you need to do this in order to perform remote deployments, even if they're to a VM on your own machine.
```

Comment:
Yeah it's public but our organization is hidden so I'm not sure how anyone would find the images. I don't think it's a security risk since we're the only people who can publish to that org and it's all open source code anyway, but just as a logistical thing it makes more sense to have a local instance. I just haven't set it up yet.